contrib/ci/coverage.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh -e
+
+# if invoked outside of CI
+if [ "$CI" != "true" ]; then
+    echo "Not running in CI"
+    exit 1
+fi
+
+gcovr -x \
+	--filter build/libfwupd \
+	--filter build/libfwupdplugin \
+	--filter build/plugins \
+	--filter build/src \
+	--exclude '^.+-test.c$' \
```

Comment:
@superm1 what's your view on self tests being in the coverage output? On one hand they're tests and not actually executed on user systems, on the other hand it's useful to know when bits of the tests are not being run.

Reply:
I think they're worth having in it, but we need to split up codecov to use "components".  Then we can have rating for plugins, for tests, for libraries, for core etc.

Reply:
https://docs.codecov.com/docs/components

Reply:
Similarly after/if we get components set up I'm not sure we should be excluding all the rustgen generated stuff.  If we're generating structs but not using them, then it is a "coverage gap".  But maybe it should be a component on it's own.